v2/options.go

```diff
@@ -0,0 +1,48 @@
+package car
+
+type ReadOptions struct {
```

Comment:
I also thought about this while writing the code :) It's certainly not necessary to expose these structs with their fields, but it also doesn't really hurt, beyond taking a bit of space in the rendered doc page.

An internal package would work, though I think it's always a bit weird when an exposed type (in this case ReadOption) references an internal type.

Reply:
> doesn't really hurt

The rationale I had with suggesting internal package here was to hide as much as possible. Otherwise, we'll be supporting it as public API for the lifetime of v2. If all things are equal, then I'd rather have the freedom to refactor this in case there is a use case that doesn't fit the `Options` abstractions proposed here.

Reply:
As per the offline chat, we've agreed to leave this as is - otherwise an internal struct type leaks in the public API.